src/MeetingsSDKAdapter.js

```diff
@@ -0,0 +1,172 @@
+import {MeetingsAdapter} from '@webex/component-adapter-interfaces';
+import {concat, from, fromEvent, Observable} from 'rxjs';
+import {filter, finalize, map, publishReplay, refCount} from 'rxjs/operators';
+
+const MEDIA_TYPE_LOCAL = 'local';
+const EVENT_MEDIA_READY = 'media:ready';
+
+/**
+ * The `MeetingsSDKAdapter` is an implementation of the `MeetingsAdapter` interface.
+ * This adapter utilizes the Webex JS SDK to create and join webex meetings.
+ *
+ * @class MeetingsSDKAdapter
+ * @extends {MeetingsAdapter}
+ */
+export default class MeetingsSDKAdapter extends MeetingsAdapter {
+  constructor(datasource) {
+    super(datasource);
+    this.getMeetingObservables = {};
+    this.meetings = {};
+  }
+
+  /**
+   * Register the SDK meeting plugin to the device
+   * and sync the meeting collection from the server.
+   */
+  async connect() {
+    await this.datasource.meetings.register();
+    await this.datasource.meetings.syncMeetings();
+  }
+
+  /**
+   * Unregister the sdk meeting plugin from the device.
+   */
+  async disconnect() {
+    await this.datasource.meetings.unregister();
+  }
+
+  /**
+   * Retrieve the local device medias and add them to the meeting
+   * with the given default media settings to the meeting.
+   *
+   * @param {string} meetingID  ID to retrieve the SDK meeting object to add the local media to
+   */
+  addLocalMedia(meetingID) {
+    const meeting = this.fetchMeeting(meetingID);
+
+    // default media setting
+    const mediaSettings = {
+      receiveVideo: true,
+      receiveAudio: true,
+      receiveShare: false,
+      sendVideo: true,
+      sendAudio: true,
+      sendShare: false,
+    };
+
+    meeting.getMediaStreams(mediaSettings).then(([localStream, localShare]) => {
```

Comment:
Let's `await` the `getMediaStreams` and `addMedia` since this is supposed to return a promise based on the [interface](https://github.com/webex/component-adapter-interfaces/blob/master/src/MeetingsAdapter.js#L102)

Reply:
the interface will change. it will not return anything. it's in the space. I will bump you to the thread